src/collectFileNames.test.ts

```diff
@@ -0,0 +1,34 @@
+import path from "node:path";
+import { describe, expect, it } from "vitest";
+
+import { collectFileNames } from "./collectFileNames.js";
+
+describe("collectFileNames", () => {
+	it("should collect files with wildcard", async () => {
+		const cwd = path.resolve(import.meta.dirname, "..");
+		const fileNames = await collectFileNames(
+			path.resolve(import.meta.dirname),
+			["*"],
+		);
+		const names = Array.isArray(fileNames)
+			? (fileNames as string[]).map((item) => item.replace(cwd, "<rootDir>"))
+			: undefined;
+
+		// Assert
+		expect(names).toContain("<rootDir>/src/collectFileNames.test.ts");
```

Comment:
[Style] IMO it's generally a code smell to use logic in modifying the results of a function under test before the assertion. The test should only happen in one way (the correct way). Instead of doing a whole ternary and `.map`, this is less extra: 

```suggestion
		expect(fileNames).toContain(`${cwd}/src/collectFileNames.test.ts`);
```